compiler/rustc_macros/src/diagnostics/mod.rs

```diff
@@ -3,6 +3,7 @@ mod diagnostic_builder;
 mod error;
 mod fluent;
 mod subdiagnostic;
+mod subdiagnostic_builder;
```

Comment:
I'm not sure why this is necessary given that the `SessionSubdiagnosticDeriveBuilder` isn't shared? `diagnostic_builder` exists because the builder is shared between `LintDiagnostic` and `SessionDiagnostic`.

Reply:
Just personal preference, I've moved it back into `subdiagnostic`.